tools/c7n_azure/tests/test_notify.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+from azure_common import BaseTest, arm_template
+
+
+class NotifyTest(BaseTest):
+    def setUp(self):
+        super(NotifyTest, self).setUp()
+
+    @arm_template('keyvault.json')
+    def test_notify_though_storage_queue(self):
+        p = self.load_policy({
+            'name': 'test-notify-for-keyvault',
+            'resource': 'azure.keyvault',
+            'actions': [
+                {'type': 'notify',
+                 'template': 'default',
+                 'priority_header': '2',
+                 'subject': 'testing notify action',
+                 'to': ['user@domain.com'],
+                 'transport':
+                     {'type': 'asq',
+                      'queue': 'https://cctstoragey6akyqpagdt3o.queue.core.windows.net/testcc'}
```

Comment:
Won't be able to rerecord this as storage names are random each time.  You can copy how I did it in `tools/c7n_azure/tests/test_storageutils.py` with the `setup_account()` which goes and finds the storage account, e.g. 

```
account = self.setup_account()
url = "https://" + account.name + ".blob.core.windows.net/testcc"
```
You can also read your queue messages (and delete them), example in that same file, e.g.

```
        # Pull messages, should be 1
        messages = StorageUtilities.get_queue_messages(*queue_settings)
        self.assertEqual(len(messages), 1)

        # Read message and delete it from queue
        for message in messages:
            self.assertEqual(message.content, u"hello queue")
            StorageUtilities.delete_queue_message(*queue_settings, message=message)
```

Reply:
Done. I moved setup_account method to the base class.